### PR TITLE
feat(sui): support utils contract deployment

### PR DESCRIPTION
**Pull request by npty**
# Description

[AXE-4848](https://axelarnetwork.atlassian.net/browse/AXE-4848)

This PR supports Utils contract deployment

[AXE-4848]: https://axelarnetwork.atlassian.net/browse/AXE-4848?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sui/deploy-contract.js [hunk 1]
```diff
@@ -66,6 +67,7 @@ const PACKAGE_CONFIGS = {
         Governance: {},
         ITS: postDeployIts,
         Squid: postDeploySquid,
+        Utils: {},
```
Comment (milapsheth) [inline sui/deploy-contract.js:70]: update readme that needs to be deployed before axelar gateway

**Review by milapsheth — APPROVED** [review]

File: sui/deploy-contract.js [hunk 2]
```diff
@@ -66,6 +67,7 @@ const PACKAGE_CONFIGS = {
         Governance: {},
         ITS: postDeployIts,
         Squid: postDeploySquid,
+        Utils: () => undefined,
```
[the remark this answers is not in the document]
Comment (npty) [reply sui/deploy-contract.js]: fixed https://github.com/axelarnetwork/axelar-contract-deployments/pull/373/commits/1e1d1babeea0f76b18b8244a0925463dafad16ea

File: sui/deploy-contract.js [hunk 3]
```diff
@@ -269,16 +271,17 @@ async function upgrade(keypair, client, supportedPackage, policy, config, chain,
     const contractConfig = contractsConfig?.[packageName];
 
     validateParameters({ isNonEmptyString: { packageName } });
+    const moveDir = `${__dirname}/move`;
```
Comment (milapsheth) [inline sui/deploy-contract.js:274]: but this dir doesn't contain a copy of move contracts, how would it work?

File: sui/deploy-contract.js [hunk 4]
```diff
@@ -269,16 +271,17 @@ async function upgrade(keypair, client, supportedPackage, policy, config, chain,
...
 
     if (packageDependencies) {
         for (const dependencies of packageDependencies) {
             const packageId = contractsConfig[dependencies]?.address;
-            updateMoveToml(dependencies, packageId);
+            updateMoveToml(dependencies.toLowerCase(), packageId, moveDir);
```
Comment (milapsheth) [inline sui/deploy-contract.js:279]: updateMoveToml is not the most elegant to read, can you refactor it to use a toml parser so it cleanly reads/writes the file?

File: sui/deploy-contract.js [hunk 5]
```diff
@@ -269,16 +271,17 @@ async function upgrade(keypair, client, supportedPackage, policy, config, chain,
     const contractConfig = contractsConfig?.[packageName];
 
     validateParameters({ isNonEmptyString: { packageName } });
+    const moveDir = `${__dirname}/move`;
 
     if (packageDependencies) {
         for (const dependencies of packageDependencies) {
```
Comment (milapsheth) [inline sui/deploy-contract.js:277]: need to split it by `,` or something to make it into an array?

Comment (milapsheth) [inline sui/deploy-contract.js:277]: can you look into doing this dynamically? read the Move.toml and get all the local dependencies, then read from config. separate PR makes sense for it

File: sui/deploy-contract.js (hunk 3 above)
Comment (npty) [reply sui/deploy-contract.js:266]: it's generated after the package deployment [here](https://github.com/axelarnetwork/axelar-contract-deployments/blob/57839b280aff4670996f2ac95218b5970f1d5e1e/sui/utils/utils.js#L63-L65)

File: sui/deploy-contract.js (hunk 5 above)
Comment (npty) [reply sui/deploy-contract.js:269]: `packageDependencies` is  a [variadic options](https://github.com/tj/commander.js?tab=readme-ov-file#variadic-option), so we don't need to transform it manually.

File: sui/deploy-contract.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (npty) [reply sui/deploy-contract.js:271]: yes, this doesn't work for `axelar-gateway` package. will do it dynamically in a separate pr as you suggested.

Comment (npty) [reply sui/deploy-contract.js:271]: will do this in a separate pr

File: sui/deploy-contract.js (hunk 5 above)
Comment (npty) [reply sui/deploy-contract.js]: add jira ticket here https://axelarnetwork.atlassian.net/browse/AXE-4977

File: sui/deploy-contract.js [hunk 6]
```diff
@@ -269,16 +271,17 @@ async function upgrade(keypair, client, supportedPackage, policy, config, chain,
     const contractConfig = contractsConfig?.[packageName];
 
     validateParameters({ isNonEmptyString: { packageName } });
+    const moveDir = `${__dirname}/move`;
 
     if (packageDependencies) {
```
[the remark this answers is not in the document]
Comment (npty) [reply sui/deploy-contract.js]: it's removed for now. will do in a separate pr

File: sui/deploy-contract.js (hunk 3 above)
Comment (milapsheth) [reply sui/deploy-contract.js]: I see

**Review by milapsheth — APPROVED** [review]

**Pull request merged**